File: scripts/fix-animejs-v4.js
const fs = require('fs');
const path = require('path');

const files = [
  'src/components/ui/Animated.js',
  'src/hooks/useAnimejs.js',
  'src/components/ui/GlassCard.js',
];

for (const file of files) {
  const fp = path.join(__dirname, '..', file);
  let content = fs.readFileSync(fp, 'utf8');
  let changed = false;
  
  // Pattern: animeAnimate({\n  targets: el,\n  ...rest\n})  →  animeAnimate(el, {\n  ...rest\n})
  // Also: animeAnimate({targets: el, ...rest}) → animeAnimate(el, {...rest})
  
  // Match animeAnimate({ followed by targets: somewhere inside
  // We need to extract the targets value and move it to first arg
  
  // Simple approach: replace animeAnimate({\n  targets: X,\n  with animeAnimate(X, {\n  
  // Multi-line targets pattern
  const mlPattern = /animeAnimate\(\{\s*\n\s*targets:\s*([^,}]+),\s*\n/g;
  if (mlPattern.test(content)) {
    content = content.replace(mlPattern, 'animeAnimate($1, {\n');
    changed = true;
  }
  
  // Single-line targets pattern: animeAnimate({targets: X, ...})
  const slPattern = /animeAnimate\(\{\s*targets:\s*([^,}]+)/g;
  // Only replace if not already converted (no first-arg pattern)
  // Count animeAnimate( patterns that look like two-arg already
  const twoArgCount = (content.match(/animeAnimate\([^\{]/g) || []).length;
  const oneArgCount = (content.match(/animeAnimate\(\{[^]*targets:/g) || []).length;
  
  if (oneArgCount > twoArgCount) {
    content = content.replace(slPattern, 'animeAnimate($1, {');
    changed = true;
  }
  
  if (changed) {
    fs.writeFileSync(fp, content, 'utf8');
    console.log(`Fixed: ${file}`);
  } else {
    console.log(`Already OK: ${file}`);
  }
}

File: scripts/fix-animejs-v4.py
#!/usr/bin/env python3
"""Fix animejs v3 → v4 API: animate({targets, ...}) → animate(targets, {...})"""
import re, sys

def fix_anime_calls(content):
    """Replace animeAnimate({targets: X, ...rest}) with animeAnimate(X, {...rest})"""
    
    def replacer(m):
        indent = m.group(1)  # whitespace before 'animeAnimate'
        targets = m.group(2).strip()
        rest = m.group(3)
        return f'{indent}animeAnimate({targets}, {{{rest}'
    
    # Pattern 1: animeAnimate({\n  targets: EXPR,\n  REST})
    # We need to match the opening call, then targets line, then rest
    result = []
    lines = content.split('\n')
    i = 0
    while i < len(lines):
        line = lines[i]
        # Look for animeAnimate({ at end of line (multi-line) or animeAnimate({ in middle
        if 'animeAnimate({' in line:
            # Check if targets: is on same line
            tm = re.search(r'^(\s*)animeAnimate\(\{\s*targets:\s*([^,}]+),(.*)$', line)
            if tm:
                # Single-line: animeAnimate({targets: X, rest})
                indent = tm.group(1)
                targets = tm.group(2).strip()
                rest = tm.group(3).strip()
                if rest.endswith('}'):
                    rest = rest[:-1].rstrip()
                    result.append(f'{indent}animeAnimate({targets}, {{{rest}}})')
                else:
                    result.append(f'{indent}animeAnimate({targets}, {{')
                    result.append(f'  {rest}')
                i += 1
                continue
            
            # Multi-line: animeAnimate({\n  targets: X,\n  rest...\n})
            tm = re.search(r'^(\s*)animeAnimate\(\{\s*$', line)
            if tm:
                indent = tm.group(1)
                # Next line should have targets:
                if i + 1 < len(lines):
                    next_line = lines[i + 1]
                    tm2 = re.match(r'^(\s*)targets:\s*([^,}]+),\s*$', next_line)
                    if tm2:
                        targets = tm2.group(2).strip()
                        result.append(f'{indent}animeAnimate({targets}, {{')
                        i += 2  # skip the targets line
                        continue
                
                # targets might be on the same line as opening brace
                tm3 = re.search(r'animeAnimate\(\{\s*targets:\s*([^,}]+),\s*$', line)
                if tm3:
                    targets = tm3.group(1).strip()
                    result.append(f'{indent}animeAnimate({targets}, {{')
                    i += 1
                    continue
        
        result.append(line)
        i += 1
    
    return '\n'.join(result)


if __name__ == '__main__':
    files = [
        'src/components/ui/Animated.js',
        'src/hooks/useAnimejs.js', 
        'src/components/ui/GlassCard.js',
    ]
    
    for f in files:
        try:
            with open(f) as fh:
                content = fh.read()
            new_content = fix_anime_calls(content)
            if new_content != content:
                with open(f, 'w') as fh:
                    fh.write(new_content)
                print(f'Fixed: {f}')
            else:
                print(f'No changes: {f}')
        except FileNotFoundError:
            print(f'Not found: {f}')
